docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,769 @@
+# Working with CHIP Tool
+
+The CHIP Tool (`chip-tool`) is a tool that allows to commission a Matter device
+into the network and to communicate with it using the Zigbee Cluster Library
+(ZCL) messages.
+
+<hr>
+
+-   [Source files](#source)
+-   [Building chip-tool](#building)
+-   [Using chip-tool for Matter accessory testing](#using)
+-   [Supported commands and options](#commands)
+
+<hr>
+
+<a name="source"></a>
+
+## Source files
+
+You can find source files of the CHIP Tool in the `examples/chip-tool`
+directory.
+
+<hr>
+
+<a name="building"></a>
+
+## Building
+
+Before you can use the `chip-tool`, you must compile it from the source on Linux
+(amd64 / aarch64) or macOS.
+
+> To ensure compatibility, build the `chip-tool` and the Matter device from the
+> same revision of the connectedhomeip repository.
+
+To build and run the `chip-tool`:
+
+1. Install all necessary packages and prepare the build system. For more
+   details, see the [Building Matter](BUILDING.md) documentation:
+
+    ```
+    sudo apt-get update
+    sudo apt-get upgrade
+
+    sudo apt-get install git gcc g++ python pkg-config libssl-dev libdbus-1-dev libglib2.0-dev libavahi-client-dev ninja-build python3-venv python3-dev python3-pip unzip libgirepository1.0-dev libcairo2-dev bluez
+    ```
+
+    If the `chip-tool` is built on a Raspberry Pi, install additional packages
+    and reboot the device:
+
+    ```
+    sudo apt-get install pi-bluetooth
+    sudo reboot
+    ```
+
+2. Clone the Project CHIP repository:
+
+    ```
+    git clone https://github.com/project-chip/connectedhomeip.git
```

Comment:
Well, the 1 step  guides how to setup the build system and get all tools required to actually build the chip-tool from sources. So, IMHO checking out the sources should be kept as a separate step. Yet another approach would be to only link the [Building Matter](BUILDING.md) document as a first step - still it may introduce some conspiracy, as it also describes how to set up the Matter repository on user's PC, so we may as well squash all first 4 steps into one which only directs to the [Building Matter](BUILDING.md).

Reply:
I'd squash to Building Matter then and add some clarifications if needed.